packages/lib/gpc/src/gpcTypes.ts

```diff
@@ -3,11 +3,32 @@ import { Identity } from "@semaphore-protocol/identity";
 import { Groth16Proof } from "snarkjs";
 
 /**
- * String specifying a named entry in a named object, in the format
- * `objectName.entryName`.  Each of the sub-parts should be a valid PODName,
- * checked by {@link POD_NAME_REGEX}.
+ * String specifying a named entry, virtual or otherwise, in a named object, in
+ * the format `objectName.entryName`.  Each of the sub-parts should be a valid
+ * PODName, checked by {@link POD_NAME_REGEX} or {@link POD_VIRTUAL_NAME_REGEX}.
+ *
+ * Examples: "ticket1.eventID", "award.$signerPublicKey"
+ */
+export type PODEntryIdentifier = `${PODName}.${PODName | PODVirtualEntryName}`;
+
+/**
+ * Regex matching legal entry identifiers for virtual POD entries; these are of
+ * the form `${PODName}.${PODVirtualEntryName}`.
+ */
+export const POD_VIRTUAL_ENTRY_IDENTIFIER_REGEX = new RegExp(
+  /([A-Za-z_]\w*)\.\$(signerPublicKey)$/
+);
+
+/**
+ * String specifying valid virtual entry name.
+ */
+export type PODVirtualEntryName = "$signerPublicKey";
```

Comment:
GitHub won't let me comment on a line too far away from the changes, but there are some more type and doc changes needed in this file:
- In GPCProofObjectConfig the `entries` key should be `PODName | PODVirtualEntryName`, and the docs should mention virtual entries.  Note that TypeScript can't really enforce this since it can't deal with the regex.
- In GPCProofEntryConfig the docs for isRevealed should talk about the different defaults for normal entries vs. for signerPublicKey.

Reply:
I actually think we should maybe discuss this further.  I think adding these "virtual" entries directly into the entries of the config, is going to be asking for trouble in the form of more special-cases like this, and will lead to confusion of what "entries" can actually contain.  What do you think of placing the entry config for $signerPublicKey in its own field of the GPC place in the GPCProofObjectConfig
```
const config: GPCProofConfig = {
  pods: {
    myPOD: {
      entries: {
        nonVirtual: { isRevealed: true },  // Default is false here
      },
      signerPublicKey: { isRevealed: true },  // Default is true here
    }
  }
} 
```

The type of the config for signerPublicKey can actually be slightly different (maybe with a shared base class) because it shouldn't include isOwnerID.  Also the documentation can make it clear where the defaults are different.

You can still use the "virtual entry" structure internally (allowing PODEntryIdentifiers to be used, including it in the entryMap, etc), but I think confusing entries for non-entries in the public interface may be too confusing to be worthwhile.  It'll make users expect these virtual entries to be present in their POD objects, etc.

Note this suggestion is explicitly the opposite of my prior thought about putting $signerPublicKey into the entries in the revealed claims.  The link between the two in my head is that virtual entries should either appear in the entries list everywhere or nowhere.